src/main/java/AssetExchangeTransaction.java

```diff
@@ -0,0 +1,114 @@
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.scalar.db.api.DistributedTransaction;
+import com.scalar.db.api.Get;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.Result;
+import com.scalar.db.config.DatabaseConfig;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.exception.transaction.TransactionException;
+import com.scalar.db.io.IntValue;
+import com.scalar.db.io.Key;
+import com.scalar.db.io.TextValue;
+import com.scalar.db.service.TransactionModule;
+import com.scalar.db.service.TransactionService;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+
+public class AssetExchangeTransaction extends AssetExchange{
+    private final TransactionService txService;
+
+    public AssetExchangeTransaction() throws IOException {
+        DatabaseConfig dbConfig = new DatabaseConfig(new FileInputStream(SCALARDB_PROPERTIES));
+        Injector injector = Guice.createInjector(new TransactionModule(dbConfig));
+        txService = injector.getInstance(TransactionService.class);
+    }
+
+    public String createAsset(String description, int price, String belong) throws TransactionException {
```

Comment:
You need to catch exceptions and call `tx.abort()` when an exception happens.

To finish a transaction, you always need to call `tx.commit()` or `tx.abort()` in the end.

Reply:
I got it. Thanks for the suggestion.